tools/src/scie_pants/update_scie_pants.py

```diff
@@ -0,0 +1,288 @@
+        if release:
+            latest_releases.append(release)
+
+    if not latest_releases:
+        log.debug(
+            f"No releases for {BINARY_NAME} compatible with {platform} found in: "
+            f"{json.dumps(releases, indent=2)}"
+        )
+        return None
+
+    # We want the highest version; so this needs to be a reverse sort.
+    latest_releases.sort(reverse=True, key=lambda rel: rel.version)
+    return latest_releases[0]
+
+
+def install_release(ptex: Ptex, release: Release, scie: Path) -> Path:
+    # The `.sha256` checksum file format is a single line with two fields, space separated.
+    # The 1st field is the hexadecimal checksum and the second field the name of the file it applies
+    # to. See: https://man7.org/linux/man-pages/man1/sha256sum.1.html
+    expected_sha256, _ = ptex.fetch_text(release.binary_sha256_url).strip().split(" ", maxsplit=1)
+
+    download_dir = Path(tempfile.mkdtemp())
+    atexit.register(shutil.rmtree, download_dir, ignore_errors=True)
+
+    binary = download_dir / release.file_name
+    with open(binary, "wb") as fp:
+        ptex.fetch_to_fp(release.binary_url, fp)
+
+    # TODO(John Sirois): Ideally this would not be 2-pass and we'd hash the download stream inline.
+    digest = hashlib.sha256()
+    with open(binary, "rb") as fp:
+        for chunk in iter(lambda: fp.read(io.DEFAULT_BUFFER_SIZE), b""):
+            digest.update(chunk)
+    actual_sha256 = digest.hexdigest()
+    if expected_sha256 != actual_sha256:
+        eol = os.linesep
+        raise ValueError(
+            f"The binary downloaded from {release.binary_url} is invalid.{eol}"
+            f"The expected fingerprint from {release.binary_sha256_url} was:{eol}"
+            f"  {expected_sha256}{eol}"
+            f"The actual fingerprint of the downloaded file is:{eol}"
+            f"  {actual_sha256}",
+        )
+
+    # Mark the binary as executable. This is needed on Unix but not on Windows, where its harmless.
+    binary.chmod(0o755)
+
+    # We first move the running scie before replacing it. This satisfies Windows constraints
+    # surrounding manipulation of running binaries.
+    backup = scie.rename(scie.with_suffix(".bak"))
+    try:
+        shutil.move(str(binary), str(scie))
+    except OSError:
+        warn(f"A backup is saved in {backup}")
+        raise
+    return backup
+
+
+def verify_release(scie: PurePath) -> str:
+    return (
+        subprocess.run(
+            args=[str(scie)],
+            env={**os.environ, "PANTS_BOOTSTRAP_VERSION": "report"},
+            stdout=subprocess.PIPE,
+            check=True,
+        )
+        .stdout.decode()
+        .strip()
+    )
+
+
+def main() -> NoReturn:
+    parser = ArgumentParser()
+    get_ptex = Ptex.add_options(parser)
+    parser.add_argument(
+        "--platform",
+        required=True,
+        # The current platform tag (<OS>-<ARCH>)
+        help=argparse.SUPPRESS,
```

Comment:
Why are these suppressed?

Reply:
They are curried by the lift manifest - user's do not pass them, so if they showed up in help that would be confusing to users.

Reply:
It hides all this garbage which the user has no change of specifying:
https://github.com/pantsbuild/scie-pants/blob/91aa50b517523fed0d054831274351e022696029/package/scie-pants.lift.json#L147-L158

Instead, they see only the unbound help:
```
$ SCIE_BOOT=update dist/scie-pants-linux-x86_64 --help
usage: tools.pex [-h] [version]

positional arguments:
  version     The version of scie-pants to update to; defaults to the latest stable version

optional arguments:
  -h, --help  show this help message and exit
```

Reply:
I need to fix that "tools.pex" prog name leak, that also means nothing to an end user, but that's for follow up.